cms/djangoapps/maintenance/views.py

```diff
@@ -0,0 +1,231 @@
+
+    template = 'maintenance/container.html'
+
+    def __init__(self, command=None):
+        self.context = {
+            'command': command if command else '',
+            'form_data': {},
+            'error': False,
+            'msg': ''
+        }
+
+    def render_response(self):
+        """ A short method to render_to_response that renders response."""
+        return render_to_response(self.template, self.context)
+
+    @method_decorator(require_global_staff)
+    def get(self, request):
+        """Render get view."""
+        return self.render_response()
+
+    def validate_course_key(self, course_key, branch=ModuleStoreEnum.BranchName.draft):
+        """
+        Validates the course_key that would be used by maintenance app views.
+
+        Arguments:
+            course_key (string): a course key
+            branch: a course locator branch, default value is ModuleStoreEnum.BranchName.draft .
+                    values can be either ModuleStoreEnum.BranchName.draft or ModuleStoreEnum.BranchName.published.
+
+        Returns:
+            course_usage_key (CourseLocator): course usage locator
+        """
+        if not course_key:
+            raise ValidationError(COURSE_KEY_ERROR_MESSAGES['empty_course_key'])
+
+        course_usage_key = CourseKey.from_string(course_key)
+
+        if not modulestore().has_course(course_usage_key):
+            raise ItemNotFoundError(COURSE_KEY_ERROR_MESSAGES['course_key_not_found'])
+
+        # get branch specific locator
+        course_usage_key = course_usage_key.for_branch(branch)
+
+        return course_usage_key
+
+
+class ForcePublishCourseView(MaintenanceBaseView):
+    """
+    View for force publishing state of the course, used by the escalation team.
+    """
+
+    def __init__(self):
+        super(ForcePublishCourseView, self).__init__(MAINTENANCE_VIEWS['force_publish_course'])
+        self.context.update({
+            'current_versions': [],
+            'updated_versions': [],
+            'form_data': {
+                'course_id': '',
+                'is_dry_run': True
+            }
+        })
+
+    @transaction.atomic
+    @method_decorator(require_global_staff)
+    def post(self, request):
+        """
+        Force publishes a course.
+
+        Arguments:
+            course_id (string): a request parameter containing course id
+            is_dry_run (string): a request parameter containing dry run value.
+                                 It is obtained from checkbox so it has either values 'on' or ''.
+        """
+
+        course_id = request.POST.get('course-id')
+        is_dry_run = bool(request.POST.get('dry-run'))
+
+        self.context.update({
+            'form_data': {
+                'course_id': course_id,
+                'is_dry_run': is_dry_run
+            }
+        })
+
+        try:
+            course_usage_key = self.validate_course_key(course_id)
+        except InvalidKeyError:
+            self.context['error'] = True
```

Comment:
I don't think so, this would be over-logging I guess. It is just a user input error.